satpy/coords_utils.py

```diff
@@ -0,0 +1,137 @@
+        # x/y coords already provided
+        return data_arr
+    if "x" not in data_arr.dims or "y" not in data_arr.dims:
+        # no defined x and y dimensions
+        return data_arr
+    if not hasattr(area, "get_proj_vectors"):
+        return data_arr
+    x, y = area.get_proj_vectors()
+
+    # convert to DataArrays
+    y_attrs = {}
+    x_attrs = {}
+
+    _check_crs_units(crs, x_attrs, y_attrs)
+
+    y = xr.DataArray(y, dims=("y",), attrs=y_attrs)
+    x = xr.DataArray(x, dims=("x",), attrs=x_attrs)
+    return data_arr.assign_coords(y=y, x=x)
+
+
+def _check_crs_units(crs, x_attrs, y_attrs):
+    if crs is None:
+        return
+    units = crs.axis_info[0].unit_name
+    # fix udunits/CF standard units
+    units = units.replace("metre", "meter")
+    if units == "degree":
+        y_attrs["units"] = "degrees_north"
+        x_attrs["units"] = "degrees_east"
+    else:
+        y_attrs["units"] = units
+        x_attrs["units"] = units
+
+
+def add_crs_xy_coords(data_arr, area):
+    """Add :class:`pyproj.crs.CRS` and x/y or lons/lats to coordinates.
+
+    For SwathDefinition or GridDefinition areas this will add a
+    `crs` coordinate and coordinates for the 2D arrays of `lons` and `lats`.
+
+    For AreaDefinition areas this will add a `crs` coordinate and the
+    1-dimensional `x` and `y` coordinate variables.
+
+    Args:
+        data_arr (xarray.DataArray): DataArray to add the 'crs'
+            coordinate.
+        area (pyresample.geometry.AreaDefinition): Area to get CRS
+            information from.
+
+    """
+    from pyresample.geometry import SwathDefinition
+
+    crs, data_arr = _add_crs(area, data_arr)
+
+    # Add x/y coordinates if possible
+    if isinstance(area, SwathDefinition):
+        # add lon/lat arrays for swath definitions
+        # SwathDefinitions created by Satpy should be assigning DataArray
+        # objects as the lons/lats attributes so use those directly to
+        # maintain original .attrs metadata (instead of converting to dask
+        # array).
+        lons = area.lons
+        lats = area.lats
+        lons.attrs.setdefault("standard_name", "longitude")
+        lons.attrs.setdefault("long_name", "longitude")
+        lons.attrs.setdefault("units", "degrees_east")
+        lats.attrs.setdefault("standard_name", "latitude")
+        lats.attrs.setdefault("long_name", "latitude")
+        lats.attrs.setdefault("units", "degrees_north")
+        # See https://github.com/pydata/xarray/issues/3068
+        # data_arr = data_arr.assign_coords(longitude=lons, latitude=lats)
+    else:
+        # Gridded data (AreaDefinition/StackedAreaDefinition)
+        data_arr = add_xy_coords(data_arr, area, crs=crs)
+    return data_arr
+
+
+def _add_crs(area, data_arr):
+    # add CRS object if pyproj 2.0+
+    try:
+        from pyproj import CRS
+    except ImportError:
+        LOG.debug("Could not add 'crs' coordinate with pyproj<2.0")
+        crs = None
```

Comment:
We now require pyproj>2.2. Do you want to remove this import check and either:

1. Move the import to the top of the module (OK with me as I see pyproj a hard requirement of satpy).
2. Make it an in-function import as it is now.

Either way, the `else:` block can be de-dented.